integration-tests/src/tests/runtime/sanity_checks.rs

```diff
@@ -231,22 +239,33 @@ fn test_sanity_used_gas() {
         .map(|bytes| u64::from_le_bytes(*bytes))
         .collect::<Vec<_>>();
 
-    // Executing `used_gas` costs `base_cost`. When executing `used_gas` twice
-    // within a metered block, the returned values should differ by that amount.
-    let base_cost =
-        node.client.read().unwrap().runtime_config.wasm_config.ext_costs.gas_cost(ExtCosts::base);
-    assert_eq!(used_gas[1] - used_gas[0], base_cost);
+    // Executing `used_gas` costs `base_cost` plus an instruction to execute the `call` itself.
+    // When executing `used_gas` twice within a metered block, the returned values should differ by
+    // that amount.
+    let runtime_config = node.client.read().unwrap().runtime_config.clone();
+    let base_cost = runtime_config.wasm_config.ext_costs.gas_cost(ExtCosts::base);
+    let op_cost = match runtime_config.wasm_config.limit_config.contract_prepare_version {
+        // Entering a new metered block, all of its instructions are paid upfront.
+        // Therefore, the difference across blocks must be larger than `base_cost`,
+        // given that the block contains other instructions besides the call of
+        // `used_gas`.
```

Comment:
This comment is very confusing to me. What does that have to do with the expected ops to be 0 for wasmer0 and wasmer2?

I see it was previously at the assertion further below. Should it be moved back there?

Reply:
I think this is the most appropriate place, given that this match block is responsible for implementing the differentiation between old vs new; I’ll see about expanding this.